app/src/main/java/com/sdp/movemeet/models/GPSPath.java

```diff
@@ -0,0 +1,147 @@
+package com.sdp.movemeet.models;
+
+import com.google.android.gms.maps.model.LatLng;
+import com.sdp.movemeet.utility.DistanceCalculator;
+
+import java.io.Serializable;
+import java.util.List;
+
+/**
+ * Represents a recorded GPS path; includes a list of recorded
+ * locations along with some calculated stats for said list
+ */
+public class GPSPath implements Serializable {
+
+    private transient List<LatLng> path;
+    private long time;
+    private float averageSpeed;
+    private float distance;
+
+    /**
+     * Returns a new GPSPath, having
+     * computed stats for the recorded GPS points.
+     * @param path
+     */
+    public GPSPath(List<LatLng> path, long time) {
+        if (path == null) throw new IllegalArgumentException();
+        if (time < 0) throw new IllegalArgumentException();
+
+        this.path = path;
+        this.time = time;
+        this.distance = -1;
+        this.averageSpeed = -1;
+    }
+
+    /**
+     * Returns the path saved in this GPSRecording as a list of coordinates
+     * @return a List<LatLng> containing all points in this path, in order
+     */
+    public List<LatLng> getPath() {
+        return this.path;
+    }
+
+    /**
+     * @return the time taken to travel over this path, in milliseconds
+     */
+    public long getTime() {
+        return this.time;
+    }
+
+    /**
+     * @return the total distance covered by this path, in meters
+     */
+    public float getDistance() {
+        if (this.distance < 0) this.distance = computeTotalDistance(this.path);
```

Comment:
I don't get the purpose of checking if the distance is less than 0. Is it because it is initialised as "-1", but then it would mean that this `getDistance()` method can be accessed only once?

Reply:
The method can be called more than once, but this makes sure that accessing it doesn't return an uninitialized value.